src/Veldrid/MTL/MTLCommandList.cs

```diff
@@ -1268,7 +1264,8 @@ private protected override void UpdateBufferCore(DeviceBuffer buffer, uint buffe
 
             lock (submittedCommandsLock)
             {
-                if (!submittedStagingBuffers.TryGetValue(cb, out var bufferList)) submittedStagingBuffers[cb] = bufferList = new List<MtlBuffer>();
+                if (!submittedStagingBuffers.TryGetValue(cb, out var bufferList))
+                    submittedStagingBuffers.Add(cb, bufferList = new List<MtlBuffer>());
```

Comment:
Hold off a bit on this PR. I want to see if I can improve this further.

Reply:
I made some ungodly changes to address this, all in the name of performance. I hope it's readable enough.